### PR TITLE
Fix UWP unit test project, add Animated module tests

### PR DESCRIPTION
**Pull request by PPatBoyd**
UWP unit test project wasn't enabled to build or run; corrected issues with each and fixed broken tests.

I added a test helper to enable synchronously waiting for async WinRT calls (e.g. get window compositor as used by Animated module).

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/2779)

### CODE REVIEW

File: vnext/ReactUWP/EndPoints/dll/react-native-uwp.x64.def [hunk 1]
```diff
@@ -81,3 +81,15 @@ nallocxWeak
 mallctlWeak
 mallctlnametomibWeak
 mallctlbymibWeak
+
+?Value@ValueAnimatedNode@uwp@react@@QEAANXZ
```
Comment (acoates-ms) [inline vnext/ReactUWP/EndPoints/dll/react-native-uwp.x64.def]: Why do all of these need to be exported? -- Is this just for UTs?  Any way we can have the UTs link in the code directly, so we can keep the exported surface small.  -- At some point we have to go over the entire exported surface and ensure it is all ABI/winrt.  So I'd like to keep us honest about what the surface is.

File: vnext/ReactWindows.sln [hunk 2]
```diff
@@ -132,17 +132,20 @@ Global
 		Chakra\Chakra.vcxitems*{c38970c0-5fbf-4d69-90d8-cbac225ae895}*SharedItemsImports = 9
 	EndGlobalSection
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|ARM = Debug|ARM
```
Comment (acoates-ms) [inline vnext/ReactWindows.sln]: I believe we have been very intentional about leaving the x64 at the top of the list. -- as that makes it the default.  the fact that saving the sln in the UI always reorders them is some very annoying VS behavior.

File: vnext/ReactWindowsCore/CreateModules.h [hunk 3]
```diff
@@ -10,7 +10,7 @@
 namespace facebook {
 namespace react {
 
-class MessageQueueueThread;
```
Comment (acoates-ms) [inline vnext/ReactWindowsCore/CreateModules.h]: nice....

File: vnext/Universal.UnitTests/App.xaml [hunk 4]
```diff
@@ -1,8 +1,7 @@
-﻿<Application
+<Application
     x:Class="ReactUWPUnitTests.App"
     xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
     xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
-    xmlns:local="using:ReactUWPUnitTests"
-    RequestedTheme="Light">
```
Comment (acoates-ms) [inline vnext/Universal.UnitTests/App.xaml]: Intentional? .. I think this was added to ensure the UI was more consistent on everyones box -- at least until we have theming more sorted.

File: vnext/Universal.UnitTests/pch.cpp [hunk 5]
```diff
@@ -1,8 +1,6 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (acoates-ms) [inline vnext/Universal.UnitTests/pch.cpp]: Should probably be keeping the copyright headers?

**Comment by PPatBoyd** [comment]
> Are these tests now run as part of the CI?

#InProgress

**Comment by PPatBoyd** [comment]
Running into build issues with size on disk; will need to split up the solutions and build more individually; likely add react-win32 to match react-uwp and add slices to build them separately while keeping the main project that points to both.

**Pull request closed without merging**